### DISCUSSION TITLE
User sign in callback

### DISCUSSION THREAD

**Discussion by szanuje** [rainbow-me/rainbowkit#904] [Q&A]
Hey, Is it possible to implement a callback when user sign in to the app? 

I'm using `rainbowkit-siwe-next-auth`.

For example, I'd like to invalidate client queries when user signed in with rainbowKit provider.

With pure next-auth I would probably use `signIn().then(() => {});` from `next-auth/react`.

**Comment by aj-may** [comment]
I implemented `onSignIn` and `onSignOut` callbacks in an authentication adapter I wrote for [UseSIWE](https://github.com/random-bits-studio/use-siwe).

https://github.com/random-bits-studio/rainbowkit-use-siwe-auth

If the team doesn't mind this approach, I would be happy to make a PR to add it to their next-auth adapter.

@markdalgleish thoughts?

**Comment by aj-may** [comment]
Actually, I already made the PR.  The change was really simple.

https://github.com/rainbow-me/rainbowkit/pull/965
